crates/bevy_pbr/src/render/skin.rs

```diff
@@ -24,17 +30,39 @@ use bevy_transform::prelude::GlobalTransform;
 /// of the GPU at runtime, which would mean not using consts anymore.
 pub const MAX_JOINTS: usize = 256;
 
+/// The total number of joints we support.
+///
+/// This is 256 GiB worth of joint matrices, which we will never hit under any
+/// reasonable circumstances.
+const MAX_TOTAL_JOINTS: u32 = 1024 * 1024 * 1024;
+
+/// The number of joints that we allocate at a time.
+///
+/// Some hardware requires that uniforms be allocated on 256-byte boundaries, so
+/// we need to allocate 4 64-byte matrices at a time to satisfy alignment
+/// requirements.
+const JOINTS_PER_ALLOCATION_UNIT: u32 = (256 / size_of::<Mat4>()) as u32;
+
+/// The maximum ratio of the number of entities whose transforms changed to the
+/// total number of joints before we re-extract all joints.
+///
+/// We use this as a heuristic to decide whether it's worth switching over to
+/// fine-grained detection to determine which skins need extraction. If the
+/// number of changed entities is over this threshold, we skip change detection
+/// and simply re-extract the transforms of all joints.
+const JOINT_EXTRACTION_THRESHOLD_FACTOR: f64 = 0.25;
```

Comment:
This heuristic intuitively makes sense to me, but I wonder if it should be exposed as configuration. Maybe something for the future with more testing to understand the balance here. I also hope that our change detection (particularly w.r.t. assets) will become cheaper in the future.

Reply:
Do you mean plugin configuration?

Reply:
No, I agree we're fine now, let's just revisit this heuristic if, e.g., asset change detection improves in the future.